### PR TITLE
Add migration service calls to migration wizard

### PR DESCRIPTION
**Pull request by kburtram**

### CODE REVIEW

File: extensions/mssql/src/mssql.d.ts [hunk 1]
```diff
@@ -495,3 +497,43 @@ export interface ISqlAssessmentService {
...
+// SqlMigration interfaces  -----------------------------------------------------------------------
+
+export interface SqlMigrationImpactedObjectInfo {
+	name: string;
+	impactDetail: string;
+	objectType: string;
+}
+
+export interface SqlMigrationAssessmentResultItem {
+	rulesetVersion: string;
```
Comment (aaomidi) [inline extensions/mssql/src/mssql.d.ts:515]: Are all of these always defined? Is there a chance any of them could be undefined?

File: extensions/mssql/src/sqlMigration/sqlMigrationService.ts [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+			initialize(): void {
+			}
+		};
+	}
+
+	private constructor(context: AppContext, protected readonly client: SqlOpsDataClient) {
+		context.registerService(constants.SqlMigrationService, this);
+	}
+
+	async getAssessments(ownerUri: string): Promise<mssql.SqlMigrationAssessmentResult> {
```
Comment (aaomidi) [inline extensions/mssql/src/sqlMigration/sqlMigrationService.ts:32]: `Promise<mssql.SqlMigrationAssessmentResult|undefined>`

File: extensions/sql-migration/src/wizard/wizardController.ts [hunk 3]
```diff
@@ -4,6 +4,7 @@
  *--------------------------------------------------------------------------------------------*/
 import * as azdata from 'azdata';
 import * as vscode from 'vscode';
+import * as mssql from '../../../mssql';
```
Comment (aaomidi) [inline extensions/sql-migration/src/wizard/wizardController.ts:7]: We should expose this in a different way, I'll work on that once the PR is merged. Not a fan of doing cross extension imports.

File: extensions/sql-migration/src/wizard/assessmentResultsDialog.ts [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+	protected async updateModel(): Promise<void> {
+		return undefined;
+	}
+
+	protected async initializeDialog(dialog: azdata.window.Dialog): Promise<void> {
+		dialog.registerContent(async view => {
+			this.assessmentTable = view.modelBuilder.table()
+				.withProperties({
+					columns: [
+						'Target',
```
Comment (aaomidi) [inline extensions/sql-migration/src/wizard/assessmentResultsDialog.ts:37]: I'm guessing we'll localize these later?

**Review by aaomidi — APPROVED** [review]

File: extensions/sql-migration/src/wizard/assessmentResultsDialog.ts (hunk 4 above)
Comment (kburtram) [reply extensions/sql-migration/src/wizard/assessmentResultsDialog.ts:37]: these are just placeholders.  the actual UX design won't have these exact strings, so no need to add localized constants now.

File: extensions/mssql/src/mssql.d.ts (hunk 1 above)
Comment (kburtram) [reply extensions/mssql/src/mssql.d.ts:515]: I'll double-check and add `|undefined` if any can be null.  IIRC they should be all defined.

File: extensions/sql-migration/src/wizard/wizardController.ts (hunk 3 above)
Comment (kburtram) [reply extensions/sql-migration/src/wizard/wizardController.ts:7]: sounds good.  let's change that in the next PR. 😄

**Pull request merged**